tests/common/devices/onyx.py

```diff
@@ -196,13 +196,41 @@ def get_speed(self, interface_name):
         if 'failed' in show_int_result and show_int_result['failed']:
             logger.error('Failed to get speed for port {} - {}'.format(interface_name, show_int_result['msg']))
             return False
-        
+
         out = show_int_result['stdout'][0].strip()
         logger.debug('Get speed for port {} from onyx: {}'.format(interface_name, out))
         if not out:
             return None
-        
+
         # The output should be something like: "Actual speed:50G"
         speed = out.split(':')[-1].strip()
         pos = speed.find('G')
         return speed[:pos] + '000'
+
+    def prepare_drop_counter_config(self, fanout_graph_facts, match_mac, set_mac, eth_field):
+        """Set configuration for drop_packets tests if fanout has onyx OS
+        Affected tests:test_equal_smac_dmac_drop, test_multicast_smac_drop
+
+        Args:
+            fanout_graph_facts (dict): fixture fanout_graph_facts
+            match_mac (str): mac address to match in openflow rule
+            set_mac (str): mac address to which match mac should be changed
+            eth_field (str): place in which replace match mac to set_mac, usually 'eth_src'
+
+        Returns:
+            boolean: True if success. Usually, the method return False only if the operation
+            is not supported or failed.
+        """
+        fanout_helper = FanoutOnyxDropCounter(self)
```

Comment:
can we share the `FanoutOnyxDropCounter` object between `prepare_drop_counter_config` and `restore_drop_counter_config`?